### PR TITLE
Enable referring to all fields/ids in offset paths

### PR DESCRIPTION
**Pull request by wemrysi**
[ch11117]

### CODE REVIEW

**Comment by wemrysi** [comment]
Released as `187.0.0-3470ed0`

File: api/src/main/scala/quasar/api/package.scala [hunk 1]
```diff
@@ -16,9 +16,13 @@
 
 package quasar
 
+import scala.List
+
 import cats.data.Kleisli
 
 package object api {
+  type DataPath = List[DataPathSegment]
```
Comment (jsantos17) [inline api/src/main/scala/quasar/api/package.scala:24]: Is this used anywhere?

**Review by jsantos17 — COMMENTED** [review]
Left some questions

File: api/src/main/scala/quasar/api/package.scala (hunk 1 above)
Comment (wemrysi) [reply api/src/main/scala/quasar/api/package.scala:24]: No, I can remove. I think I added before I remembered the offset is non-empty and didn't end up using it.

File: api/src/main/scala/quasar/api/DataPathSegment.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          case AllIndices => true
+          case _ => false
+        }
+      }
+    }
+
+  implicit val dataPathSegmentShow: Show[DataPathSegment] =
+    Show show {
+      case Field(n) => s".$n"
+      case AllFields => "{*}"
```
[the remark this answers is not in the document]
Comment (wemrysi) [reply api/src/main/scala/quasar/api/DataPathSegment.scala:61]: Good question, I hadn't considered them, not sure if anyone else besides you has 😄 We know we want values for sure, so we could always add keys in the future.

**Review by jsantos17 — APPROVED** [review]

**Pull request merged**